File: Core-backend-layer/Programming-fundamentals/Java/OOP/src/ClassCasting/GroceryProduct.java
class GroceryProduct extends Product {
    int quantityInStock;

    public GroceryProduct(String name, double price, String category, int quantityInStock) {
        super(name, price, category);
        this.quantityInStock = quantityInStock;
    }

    public void displayInfo() {
        super.displayInfo();
        System.out.println("Quantity in Stock: " + quantityInStock);
    }

    public void restock(int quantity) {
        System.out.println("Restocking " + quantity + " units of " + name);
        this.quantityInStock += quantity;
    }
}

File: Core-backend-layer/Programming-fundamentals/Python/OOP/src/class_casting/grocery_product.py
from product import Product

class GroceryProduct(Product):
    def __init__(self, name, price, category, quantity_in_stock):
        super().__init__(name, price, category)
        self.quantity_in_stock = quantity_in_stock

    def display_info(self):
        super().display_info()
        print(f"Quantity in Stock: {self.quantity_in_stock}")

    def restock(self, quantity):
        print(f"Restocking {quantity} units of {self.name}")
        self.quantity_in_stock += quantity